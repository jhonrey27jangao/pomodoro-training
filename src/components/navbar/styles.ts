import styled from "styled-components";

export const StyledHeader = styled.header`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 3.5rem;
  line-height: 3.5rem;
  background-color: #ffffff;
  box-shadow: 0 0 0.5rem 0 rgba(161, 177, 204, 0.325);
  cursor: default;
  z-index: 10000;

  & a {
    text-decoration: none;
    color: #7a7d86;
  }

  & a:hover {
    color: #e59dc4;
  }
`;

export const StyledLogo = styled.div`
  position: absolute;
  top: 0;
  left: 2rem;
  text-transform: uppercase;

  & a {
    display: block;
    height: inherit;
    line-height: inherit;
    border-bottom: 0;
  }

  & a strong {
    font-weight: 900;
  }
`;

export const StyledNav = styled.nav`
  position: absolute;
  top: 0;
  right: 2rem;

  & ul {
    list-style: none;
    padding: 0;
    margin: 0;
  }

  & ul li {
    position: relative;
    display: inline-block;
    margin: 1px 0.75rem;
    padding: 0;
  }

  & ul li a {
    padding: 0 0.75rem;
    font-weight: 600;
    overflow: hidden;
  }

  & ul li.current:after {
    content: "";
    display: block;
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 2px;
    background-image: linear-gradient(90deg, #a480ff, #ff8f83);
  }
`;
