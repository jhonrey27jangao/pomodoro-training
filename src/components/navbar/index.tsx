import React from "react";
import { StyledHeader, StyledLogo, StyledNav } from "./styles";

export const Navbar: React.FC = () => {
  return (
    <StyledHeader>
      <StyledLogo>
        <a href="https://steamuserimages-a.akamaihd.net/ugc/946220022126942778/F7EA1482B7967952540BB8F0CEE0822048BDE466/">
          <strong>Dont click here!</strong> Please?
        </a>
      </StyledLogo>
      <StyledNav>
        <ul>
          <li>
            <a href="https://assets.change.org/photos/3/wn/ua/SnwNuAfdCHjPqqi-800x450-noPad.jpg?1554888818">Home</a>
          </li>
          <li className="current">
            <a href="https://i.kym-cdn.com/photos/images/newsfeed/001/443/311/70f.gif">Pomodoro</a>
          </li>
          <li>
            <a href="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTTAUTKagxjPG_3C_tbizZOkGZtqiLCYMcVpDTBECW7Ht83jpG_">About Mar</a>
          </li>
          <li>
            <a href="https://img-comment-fun.9cache.com/media/aQMBM88/ajGMNMAl_700w_0.jpg">Roxas Shitposting</a>
          </li>
        </ul>
      </StyledNav>
    </StyledHeader>
  );
};
