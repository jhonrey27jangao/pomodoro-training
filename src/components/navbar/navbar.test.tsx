import React from 'react'
import {render} from '@testing-library/react';
import {Navbar} from './navbar'

describe('Test for Navbar Component', () => {
    it('should render Navbar', () => {
        const {container} = render(
            <Navbar/>
        )
        expect(container).toMatchSnapshot();
    })

})