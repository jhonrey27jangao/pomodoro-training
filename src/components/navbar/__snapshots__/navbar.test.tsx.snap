// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Navbar Component should render Navbar 1`] = `
<div>
  <header
    class="sc-AxjAm cqMWhk"
  >
    <div
      class="sc-AxirZ glxPCh"
    >
      <a
        href="https://steamuserimages-a.akamaihd.net/ugc/946220022126942778/F7EA1482B7967952540BB8F0CEE0822048BDE466/"
      >
        <strong>
          Dont click here!
        </strong>
         Please?
      </a>
    </div>
    <nav
      class="sc-AxiKw jXRZve"
    >
      <ul>
        <li>
          <a
            href="https://assets.change.org/photos/3/wn/ua/SnwNuAfdCHjPqqi-800x450-noPad.jpg?1554888818"
          >
            Home
          </a>
        </li>
        <li
          class="current"
        >
          <a
            href="https://i.kym-cdn.com/photos/images/newsfeed/001/443/311/70f.gif"
          >
            Pomodoro
          </a>
        </li>
        <li>
          <a
            href="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTTAUTKagxjPG_3C_tbizZOkGZtqiLCYMcVpDTBECW7Ht83jpG_"
          >
            About Mar
          </a>
        </li>
        <li>
          <a
            href="https://img-comment-fun.9cache.com/media/aQMBM88/ajGMNMAl_700w_0.jpg"
          >
            Roxas Shitposting
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
